File: src/cryptographie/c/aes.c
#include <stdio.h>
#include <string.h>

char* encrypt(char* text, char* key) {
    int textLength = strlen(text);
    char* encryptedText = (char*)malloc(textLength + 1);

    for (int i = 0; i < textLength; i++) {
        if (isalpha(text[i])) {
            char isUpperCase = isupper(text[i]);
            text[i] = tolower(text[i]);
            int charIndex = text[i] - 'a';
            char encryptedChar = isUpperCase ? toupper(key[charIndex]) : key[charIndex];
            encryptedText[i] = encryptedChar;
        } else {
            encryptedText[i] = text[i];
        }
    }
    encryptedText[textLength] = '\0';
    return encryptedText;
}

char* decrypt(char* encryptedText, char* key) {
    int textLength = strlen(encryptedText);
    char* decryptedText = (char*)malloc(textLength + 1);

    for (int i = 0; i < textLength; i++) {
        if (isalpha(encryptedText[i])) {
            char isUpperCase = isupper(encryptedText[i]);
            encryptedText[i] = tolower(encryptedText[i]);
            char* charIndex = strchr(key, encryptedText[i]);
            int index = charIndex - key;
            char decryptedChar = index + 'a';
            decryptedChar = isUpperCase ? toupper(decryptedChar) : decryptedChar;
            decryptedText[i] = decryptedChar;
        } else {
            decryptedText[i] = encryptedText[i];
        }
    }
    decryptedText[textLength] = '\0';
    return decryptedText;
}

int main() {
    char key[] = "bcdefghijklmnopqrstuvwxyza";
    char text[] = "Hello, World!";
    char* encryptedText = encrypt(text, key);
    printf("Texte chiffré : %s\n", encryptedText);
    char* decryptedText = decrypt(encryptedText, key);
    printf("Texte déchiffré : %s\n", decryptedText);

    free(encryptedText);
    free(decryptedText);

    return 0;
}

File: src/cryptographie/python/aes.py
def encrypt(text, key):
    encrypted_text = ""
    for char in text:
        if char.isalpha():
            is_upper = char.isupper()
            char = char.lower()
            char_index = ord(char) - ord('a')
            encrypted_char = key[char_index].upper() if is_upper else key[char_index]
            encrypted_text += encrypted_char
        else:
            encrypted_text += char
    return encrypted_text

def decrypt(encrypted_text, key):
    decrypted_text = ""
    for char in encrypted_text:
        if char.isalpha():
            is_upper = char.isupper()
            char = char.lower()
            char_index = key.index(char)
            decrypted_char = chr(char_index + ord('a')).upper() if is_upper else chr(char_index + ord('a'))
            decrypted_text += decrypted_char
        else:
            decrypted_text += char
    return decrypted_text

# Exemple d'utilisation
key = "bcdefghijklmnopqrstuvwxyza"
text = "Hello, World!"
encrypted_text = encrypt(text, key)
print("Texte chiffré :", encrypted_text)
decrypted_text = decrypt(encrypted_text, key)
print("Texte déchiffré :", decrypted_text)
